Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/model/Tasks.java

```diff
@@ -53,9 +53,13 @@ public boolean isEmpty() {
         return internalMap().isEmpty();
     }
 
-    public boolean isMostRecentAction(SyncedData<Task> syncedData) {
-        SyncedData<Task> currentSyncData = internalMap().get(syncedData.data().id());
-        return currentSyncData == null || syncedData.lastSyncAction() >= currentSyncData.lastSyncAction();
+    public boolean hasNoTaskMoreRecentThan(SyncedData<Task> syncedData) {
+        return hasNoTaskMoreRecentThan(syncedData.data().id(), syncedData.lastSyncAction());
+    }
+
+    public boolean hasNoTaskMoreRecentThan(Id id, long lastSyncAction) {
```

Comment:
The name is a bit confusing, it does not check that it does not have any Task more recent than the given timestamp. Rather that there is no action on the given task that is more recent than the given timestamp.